### PR TITLE
DP-18737 CSV table paragraph.

### PR DESCRIPTION
**Pull request by oknate**
**Description:**
This PR adds a CSV file paragraph type to the the Information Detail page's Information details section.

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-18737

**To Test:**
- [ ] run `composer install`
- [ ] run `drush cim`
- [ ] run `drush cr -y`
- [ ] Go to the edit section of an Information Detail page, such as http://mass.local/info-details/vaccine-resource-guide-for-employees-and-employers/edit
- [ ] Go to the content sections and add a "CSV File" paragraph to one of the sections. (You can use this file)
[olympic-winners.csv](https://github.com/massgov/openmass/files/6634974/olympic-winners.csv)
- [ ] save and view
- [ ] go back to edit and toggle off paging and save
- [ ] go back to edit and toggle on paging and save
- [ ] go back to edit and toggle off searching and save
- [ ] go back to edit and toggle on searching and save
- [ ] go back to edit and toggle on lengthChange and save
- [ ] go back to edit and toggle off lengthChange and save
- [ ] go back to edit and toggle on download and save (will show a download link when rendered).
- [ ] go back to edit and toggle off download and save
- [ ] go back to test the pageLength setting with a few options and make sure the number of rows on initial load match.
- [ ] go back and edit and test the two responsive options

https://www.youtube.com/watch?v=6GBSoYuk-IU

csv_field_editorial_ui

csv_field_rendered

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

**Comment by oknate** [comment]
I updated the PR.  Some of the configs needed update after I fixed missing schema for the csv_field contrib module.  This was very minor, it was using integers instead of booleans for a few of the configs.

**Comment by clairesunstudio** [comment]
Link to Mayflower PR https://github.com/massgov/mayflower/pull/1439

Two outstanding issues:
1. When table is set to x-large width, there's a width style set on the table, causing it to not expand to the full width
2. In the [datatables child row example](https://datatables.net/examples/api/row_details.html), the toggle icon is in a different column. But in the prototype, it's in the first column.

**Comment by clairesunstudio** [comment]
Note! child row controls are not keyboard accessible in datatables examples (A11Y)

https://datatables.net/examples/api/row_details.html

https://datatables.net/extensions/responsive/examples/child-rows/whole-row-control.html

**Comment by clairesunstudio** [comment]
Test page and examples created on feature2 https://massgovfeature2.prod.acquia-sites.com/info-details/test-data-tables---unpublished

**Comment by clairesunstudio** [comment]
> Note! child row controls are not keyboard accessible in datatables examples (A11Y)
> 
> https://datatables.net/examples/api/row_details.html
> 
> https://datatables.net/extensions/responsive/examples/child-rows/whole-row-control.html

Update: @ygannett was able to control the toggle button in the example using VO

**Comment by clairesunstudio** [comment]
https://www.drupal.org/project/csv_field/releases/1.0.0-beta2

**Review by tom-fleming — APPROVED** [review]

**Review by tom-fleming — APPROVED** [review]

File: docroot/themes/custom/mass_theme/templates/paragraphs/paragraph--csv-table.html.twig [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  '#type':   'processed_text',
+  '#text':    paragraph.field_csvtable_caption.value ?: '',
+  '#format':  paragraph.field_csvtable_caption.format ?: 'basic_html',
+} %}
+
+{% include "@atoms/09-media/figure.twig" with {
+  figure: {
+    'title': {
+      'text': paragraph.field_csvtable_title.value,
+      'visible': true,
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/templates/paragraphs/paragraph--csv-table.html.twig:49]:
```suggestion
      'level': 3,
```

File: docroot/themes/custom/mass_theme/templates/paragraphs/paragraph--csv-table.html.twig [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    'title.visible': true,
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/templates/paragraphs/paragraph--csv-table.html.twig]:
```suggestion
```

File: docroot/themes/custom/mass_theme/templates/paragraphs/paragraph--csv-table.html.twig [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  '#type':   'processed_text',
+  '#text':    paragraph.field_csvtable_caption.value ?: '',
+  '#format':  paragraph.field_csvtable_caption.format ?: 'basic_html',
+} %}
+
+{% include "@atoms/09-media/figure.twig" with {
+  figure: {
+    'title': {
+      'text': paragraph.field_csvtable_title.value,
+      'level': 3,
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/templates/paragraphs/paragraph--csv-table.html.twig]:
```suggestion
      'visible: true,
```

File: docroot/themes/custom/mass_theme/templates/paragraphs/paragraph--csv-table.html.twig [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  '#type':   'processed_text',
+  '#text':    paragraph.field_csvtable_caption.value ?: '',
+  '#format':  paragraph.field_csvtable_caption.format ?: 'basic_html',
+} %}
+
+{% include "@atoms/09-media/figure.twig" with {
+  figure: {
+    'title': {
+      'text': paragraph.field_csvtable_title.value,
+      'visible: true,
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/templates/paragraphs/paragraph--csv-table.html.twig]:
```suggestion
      'visible: true,
      'level': 3
```

**Comment by clairesunstudio** [comment]
php unit tests are failing on the same tests as develop, will be fixed after develop merging in #846

**Pull request merged**